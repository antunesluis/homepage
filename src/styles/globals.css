@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  /* Light theme */
  --color-background: #fafafa;
  --color-foreground: #18181b;

  /* Primary colors - Blue/Indigo palette */
  --color-primary: #4f46e5;
  --color-primary-hover: #4338ca;
  --color-primary-muted: #e0e7ff;

  /* Secondary colors - Teal palette */
  --color-secondary: #0d9488;
  --color-secondary-hover: #0f766e;
  --color-secondary-muted: #ccfbf1;

  /* Accent colors */
  --color-accent: #f43f5e;
  --color-accent-hover: #e11d48;
  --color-accent-muted: #ffe4e6;

  /* Gray scale */
  --color-muted: #f4f4f5;
  --color-border: #e4e4e7;
  --color-border-hover: #d4d4d8;

  /* Surface colors for cards and elevated elements */
  --color-surface: #ffffff;
  --color-surface-hover: #f8fafc;
}

.dark {
  /* Dark theme */
  --color-background: #09090b;
  --color-foreground: #fafafa;

  /* Primary colors - adjusted for dark mode */
  --color-primary: #6366f1;
  --color-primary-hover: #818cf8;
  --color-primary-muted: #1e1b4b;

  /* Secondary colors - adjusted for dark mode */
  --color-secondary: #14b8a6;
  --color-secondary-hover: #2dd4bf;
  --color-secondary-muted: #134e4a;

  /* Accent colors - adjusted for dark mode */
  --color-accent: #fb7185;
  --color-accent-hover: #fda4af;
  --color-accent-muted: #4c1d24;

  /* Gray scale - darker variants */
  --color-muted: #27272a;
  --color-border: #3f3f46;
  --color-border-hover: #52525b;

  /* Surface colors for dark mode */
  --color-surface: #18181b;
  --color-surface-hover: #27272a;
}

body {
  @apply bg-background text-foreground;
  transition:
    background-color 0.3s ease-in-out,
    color 0.3s ease-in-out;
}
