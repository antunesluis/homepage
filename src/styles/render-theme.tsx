import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'next-themes';

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider attribute="class" defaultTheme="light" enableSystem={false}>
      {children}
    </ThemeProvider>,
  );
};
