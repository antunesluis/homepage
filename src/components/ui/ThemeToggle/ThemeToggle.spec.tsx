import { screen } from '@testing-library/react';
import { renderTheme } from '@/styles/render-theme';
import { ThemeToggle } from '.';

describe('<ThemeToggle />', () => {
  it('should render theme toggle button', () => {
    renderTheme(<ThemeToggle />);
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('should match the snapshot', () => {
    const { container } = renderTheme(<ThemeToggle />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
