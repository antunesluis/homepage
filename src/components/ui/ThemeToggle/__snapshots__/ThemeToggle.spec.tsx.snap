// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ThemeToggle /> > should match the snapshot 1`] = `
<button
  class="inline-flex items-center justify-center font-medium transition-colors duration-200 select-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 active:translate-y-[1px] hover:bg-muted bg-transparent text-sm gap-1.5 relative w-9 h-9 px-0 rounded-lg"
  title="Ativar modo escuro"
  type="button"
>
  <svg
    class="lucide lucide-sun h-5 w-5 transition-transform"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="4"
    />
    <path
      d="M12 2v2"
    />
    <path
      d="M12 20v2"
    />
    <path
      d="m4.93 4.93 1.41 1.41"
    />
    <path
      d="m17.66 17.66 1.41 1.41"
    />
    <path
      d="M2 12h2"
    />
    <path
      d="M20 12h2"
    />
    <path
      d="m6.34 17.66-1.41 1.41"
    />
    <path
      d="m19.07 4.93-1.41 1.41"
    />
  </svg>
  <span
    class="sr-only"
  >
    Ativar modo escuro
  </span>
</button>
`;
